packages/base/Slider/src/Slider/Slider.tsx

```diff
@@ -43,33 +31,47 @@ export interface Props extends ComponentProps<typeof MUISlider> {
   tooltip?: ValueLabelDisplay
   /** The format function the value tooltip's value. */
   tooltipFormat?: string | ((value: number, index: number) => React.ReactNode)
-  /** Show a compact tooltip */
-  compact?: boolean
-  /** Disable the portal behavior of the tooltip. The children stay within it's parent */
-  disablePortal?: boolean
   /** Callback invoked when slider changes its state. */
-  onChange?: (event: ChangeEvent<{}>, value: Value) => void
+  onChange?: (event: Event, value: Value, activeThumb: number) => void
   /** Hide thumb when value is undefined or null. Works only when the component is controlled. */
   hideThumbOnEmpty?: boolean
   /** Disable track highlight. */
   disableTrackHighlight?: boolean
 }
 
-const createDefaultValueLabelComponent = (
-  tooltip?: ValueLabelDisplay,
-  disablePortal?: boolean,
-  compact?: boolean
-) => {
-  const ValueLableComponent = (props: ValueLabelProps) => (
-    <SliderValueLabel
-      {...props}
-      tooltip={tooltip}
-      disablePortal={disablePortal}
-      compact={compact}
-    />
-  )
+const createCustomMark = (hideTrack?: boolean, hideThumb?: boolean) => {
+  const Mark = ({
+    markActive,
+    ownerState,
+    'data-index': dataIndex,
+    style,
+  }: SliderMarkSlotProps) => {
+    return (
+      <span
+        data-index={dataIndex}
+        style={style}
+        className={twJoin(
+          'absolute w-[6px] h-[6px] rounded-[50%] border-[2px] border-solid border-white opacity-100 top-[1.5px] -translate-x-2/4 box-content',
+          hideThumb &&
+            markActive &&
+            ownerState.value === undefined &&
+            dataIndex === 0 &&
+            'bg-gray-500',
+          markActive && !hideTrack ? 'bg-blue-500' : 'bg-gray-500'
+        )}
+      />
+    )
+  }
+
+  return Mark
+}
+
+const flexCommonStyles = 'justify-center items-center'
 
-  return ValueLableComponent
+const tooltipStates: Record<ValueLabelDisplay, string> = {
+  off: 'hidden',
+  auto: twJoin('hidden group-hover/thumb:flex', flexCommonStyles),
+  on: twJoin('flex', flexCommonStyles),
 }
```

Comment:
No need for `twJoin` here, we are just joining strings after all, no conditionals.
Also to keep it simple, and avoid complexity by indirection, I would just copy and paste the styles, they are the same _for now_ , but they might diverge in the future, [no abstraction is always better than the wrong abstraction](https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction)

```suggestion
const tooltipStates: Record<ValueLabelDisplay, string> = {
  off: 'hidden',
  auto: 'hidden group-hover/thumb:flex justify-center items-center',
  on: 'flex justify-center items-center',
}
```